### PR TITLE
feat(edgenetwork/v1): generate library

### PR DESCRIPTION
**Pull request by alevenberg**
PiperOrigin-RevId: 579982620
b/309532520

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13089)

### CODE REVIEW

File: ci/abi-dumps/google_cloud_cpp_accessapproval.expected.abi.dump.gz
Comment (coryan) [inline ci/abi-dumps/google_cloud_cpp_accessapproval.expected.abi.dump.gz:1]: You should not need to update all the ABI dumps, only the one for the new library?

File: google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc [hunk 1]
```diff
@@ -92,14 +92,10 @@ FirewallPoliciesRestConnectionImpl::AddAssociation(
       },
       [request](std::string const& op,
                 google::cloud::cpp::compute::global_organization_operations::
-                    v1::GetOperationRequest& r) {
-        r.set_operation(op);
-      },
+                    v1::GetOperationRequest& r) { r.set_operation(op); },
```
Comment (coryan) [inline google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc]: nit: something went wrong in your sequence of commits (why change compute?). But the final result looks Okay. So 🤷

File: google/cloud/edgenetwork/quickstart/quickstart.cc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include "google/cloud/location.h"
+#include <iostream>
+
+int main(int argc, char* argv[]) try {
+  if (argc != 3) {
+    std::cerr << "Usage: " << argv[0] << " project-id location-id\n";
+    return 1;
+  }
+
+  auto const location = google::cloud::Location(argv[1], absl::StrCat(argv[2]));
```
Comment (coryan) [inline google/cloud/edgenetwork/quickstart/quickstart.cc:26]: I think we can remove the `absl::StrCat()`?

File: ci/abi-dumps/google_cloud_cpp_accessapproval.expected.abi.dump.gz
Comment (alevenberg) [reply ci/abi-dumps/google_cloud_cpp_accessapproval.expected.abi.dump.gz:1]: Agh you are right. I'll fix it. I was multitasking and then the env variable got unset (i think)

File: google/cloud/edgenetwork/quickstart/README.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+set GRPC_DEFAULT_SSL_ROOTS_FILE_PATH=%cd%\roots.pem
+```
+
+[authentication-quickstart]: https://cloud.google.com/docs/authentication/getting-started "Authentication Getting Started"
+[bazel-install]: https://docs.bazel.build/versions/main/install.html
+[choco-cmake-link]: https://chocolatey.org/packages/cmake
+[grpc-roots-pem-bug]: https://github.com/grpc/grpc/issues/16571
+[homebrew-cmake-link]: https://formulae.brew.sh/formula/cmake
+[howto-setup-dev-workstation]: /doc/contributor/howto-guide-setup-development-workstation.md
+[quickstart-link]: https://cloud.google.com/edgenetwork/docs/quickstart
```
Comment (coryan) [inline google/cloud/edgenetwork/quickstart/README.md:155]: This is a 404, consider something from https://cloud.google.com/distributed-cloud/edge/latest/docs/overview

File: google/cloud/edgenetwork/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+## More Information
+
+- Official documentation about the
+  [Distributed Cloud Edge Network API][cloud-service-docs] service
+- [Reference doxygen documentation][doxygen-link] for each release of this
+  client library
+- Detailed header comments in our [public `.h`][source-link] files
+
+[cloud-service-docs]: https://cloud.google.com/edgenetwork
```
Comment (coryan) [inline google/cloud/edgenetwork/README.md:58]: This is a 404, maybe use the link from the .json file?

File: ci/abi-dumps/google_cloud_cpp_accessapproval.expected.abi.dump.gz
Comment (alevenberg) [reply ci/abi-dumps/google_cloud_cpp_accessapproval.expected.abi.dump.gz:1]: Fixed

**Review by coryan — CHANGES_REQUESTED** [review]
A few nits around broken links and maybe some remnants in the quickstart.

File: google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc]: Yes something did go wrong, I ran checkers twice I think... and added it as a fixup commit

File: google/cloud/edgenetwork/quickstart/quickstart.cc (hunk 2 above)
Comment (alevenberg) [reply google/cloud/edgenetwork/quickstart/quickstart.cc:26]: Good catch. Done.

File: google/cloud/edgenetwork/quickstart/README.md (hunk 3 above)
Comment (alevenberg) [reply google/cloud/edgenetwork/quickstart/README.md:155]: Fixed.

File: google/cloud/edgenetwork/README.md (hunk 4 above)
Comment (alevenberg) [reply google/cloud/edgenetwork/README.md:58]: Done

File: google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc [hunk 5]
```diff
@@ -92,10 +92,14 @@ FirewallPoliciesRestConnectionImpl::AddAssociation(
       },
       [request](std::string const& op,
                 google::cloud::cpp::compute::global_organization_operations::
-                    v1::GetOperationRequest& r) { r.set_operation(op); },
+                    v1::GetOperationRequest& r) {
+        r.set_operation(op);
+      },
```
Comment (devbww) [inline google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc]: I'm not sure why these re-formattings are getting flipped back-and-forth, but, given that it is unrelated to adding the new library, I think I would revert this file and figure things out after.

File: google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc]: Ahh ... I guess this negates my previous comment.

File: google/cloud/edgenetwork/CMakeLists.txt [hunk 6]
```diff
@@ -16,7 +16,7 @@
 
 include(GoogleCloudCppLibrary)
 
-set(GOOGLE_CLOUD_CPP_SERVICE_DIRS "v1/")
+set(GOOGLE_CLOUD_CPP_SERVICE_DIRS "" "v1/")
```
Comment (devbww) [inline google/cloud/edgenetwork/CMakeLists.txt:19]: Why did we need to add `""`?

**Review by coryan — APPROVED** [review]

File: google/cloud/edgenetwork/CMakeLists.txt (hunk 6 above)
Comment (alevenberg) [reply google/cloud/edgenetwork/CMakeLists.txt:19]: For the mocks. You need it for the CI to pass. edgecompute is the same.

File: generator/generator_config.textproto [hunk 7]
```diff
@@ -2034,6 +2034,15 @@ service {
   retryable_status_codes: ["kUnavailable"]
 }
 
+# Edge network
```
[the remark this answers is not in the document]
Comment (alevenberg) [reply generator/generator_config.textproto:2037]: https://github.com/googleapis/google-cloud-cpp/pull/13096

File: google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc (hunk 5 above)
Comment (alevenberg) [reply google/cloud/compute/firewall_policies/v1/internal/firewall_policies_rest_connection_impl.cc]: Missed this one sorry. I can revert tomorrow

File: google/cloud/edgenetwork/CMakeLists.txt (hunk 6 above)
Comment (devbww) [reply google/cloud/edgenetwork/CMakeLists.txt:19]: Hmmm.  It sounds like we should not have set `forwarding_product_path: "google/cloud/edgenetwork"`.  It should only be used for backwards compatibility with a previous, non-versioned release of the library.  It may be worth regenerating if that is the case.
